docs/user-guides/client-api-config.md

```diff
@@ -0,0 +1,676 @@
+Please select the section which you need.
+
+Please also see each API document.
+
+## Insert Service
+
+`Insert` is inserting new vectors into the Vald cluster.
+It requires the vector, its ID (specific ID for the vector), and optional configuration.
+
+### Configuration
+
+```rpc
+// Represent insert configuration.
+message Config {
+  // Check whether or not the same set of vector and ID is already inserted.
+  bool skip_strict_exist_check = 1;
+  // Configuration for filters if your Vald cluster uses filters.
+  Filter.Config filters = 2;
+  // The timestamp when the vector was inserted.
+  int64 timestamp = 3;
+}
+```
+
+<details><summary>Insert Configuration Sample (Go)</summary><br>
+
+```go
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald-client-go/v1/payload"
+	"github.com/vdaas/vald-client-go/v1/vald"
+	"google.golang.org/grpc"
+)
+
+func main() {
+	// Create connection
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	target := "localhost:8080"
+	conn, err := grpc.DialContext(ctx, target)
+	if err != nil {
+		panic(err)
+	}
+	// Init Vald client
+	client := vald.NewValdClient(conn)
+
+	// Insert sample
+	location, err := client.Insert(ctx, &payload.Insert_Request{
+		// Vector information (mandatory)
+		Vector: &payload.Object_Vector{
+			// Set the vector
+			Vector: []float32{0, 1, 2},
+			// Set the specific ID for the vector
+			// The ID must not be indexed.
+			Id:     "sample",
+		},
+		// Insert configuration (optional)
+		Config: &payload.Insert_Config{
+			SkipStrictExistCheck: false,
+			Filters: &payload.Filter_Config{
+				Targets: []*payload.Filter_Target{
+					{
+						Host: "vald-ingress-filter",
+						Port: 8081,
+					},
+				},
+			},
+			Timestamp: time.Now().UnixMilli(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	...
+}
+```
+
+</details>
+
+#### skip_strict_exist_check
+
+`skip_strict_exist_check` is the flag for checking whether the same set of the vector and ID is already inserted or not.
+If it is set as `true`, the checking function will be skipped.<BR>
+The default value is `false`.
+
+#### filters
+
+`filters` is the configuration when using filter functions.
+In the `Insert` section, it is popular to use ingress filtering.
+
+The detailed configuration is following.
+
+```rpc
+// Filter related messages.
+message Filter {
+
+  // Represent the target filter server.
+  message Target {
+    // The target filter component hostname.
+    string host = 1;
+    // The target filter component port.
+    uint32 port = 2;
+  }
+
+  // Represent filter configuration.
+  message Config {
+    // Represent the filter target configuration.
+    repeated Target targets = 1;
+  }
+}
+```
+
+#### timestamp
+
+`timestamp` is the timestamp when the vector is inserted.
+When `timestamp` is not set, the current time will be used.
+
+## Update Service
+
+`Update` is updating vectors already inserted in the `vald-agent` component.
```

Comment:
```suggestion
`Update` service allows user to update vector(s) which is already inserted in the `vald-agent` component.
```

Reply:
Also, do we need to specify vald-agent here?

Reply:
```suggestion
`Update` service allows users to update vector(s) which is already inserted in the Vald cluster.
```